src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/ReferenceCountedDisposableCache.cs

```diff
@@ -80,6 +83,12 @@ public void Dispose()
                 // Dispose the underlying value
                 Value.Dispose();
             }
+
+            public ValueTask DisposeAsync()
+            {
+                Dispose();
+                return ValueTaskFactory.CompletedTask;
+            }
```

Comment:
We should remove all of the `DisposeAsync()` methods that just pass through to `Dispose()`. They mislead consumers into believing that code using the types _need_ to be asynchronous for optimal dispose, even though there is no advantage.